File: ganilib-core/src/main/java/com/gani/lib/collection/SelfTruncatingMap.java
package com.gani.lib.collection;

import java.util.LinkedHashMap;

public class SelfTruncatingMap<K, V> extends LinkedHashMap<K, V> {
  private static final long serialVersionUID = 1L;
  private int maxCapacity;
  
  public SelfTruncatingMap(int maxCapacity) {
    super(maxCapacity);
    this.maxCapacity = maxCapacity;
  }

  int getMaxCapacity() {
    return maxCapacity;
  }

  @Override
  protected boolean removeEldestEntry(Entry<K, V> eldest) {
    return size() > maxCapacity;
  }
}

File: ganilib-core/src/main/java/com/gani/lib/collection/SelfTruncatingMap.kt
package com.gani.lib.collection

import java.util.*

class SelfTruncatingMap<K, V>(internal val maxCapacity: Int) : LinkedHashMap<K, V>(maxCapacity) {

    override fun removeEldestEntry(eldest: Map.Entry<K, V>): Boolean {
        return size > maxCapacity
    }

    companion object {
        private val serialVersionUID = 1L
    }
}
